test/routing_test.rb

```diff
@@ -660,6 +660,18 @@ class RoutingTest < Minitest::Test
     assert ok?
   end
 
+  it 'unescapes named parameters and splats' do
+    mock_app {
+      get '/:foo/*' do |a, b|
+        assert_equal "foo\x80", params['foo']
```

Comment:
Here we should also do:

```ruby
assert_predicate params['foo'], :valid_encoding?
```

I suspect it will fail.

Reply:
You were right on, this assertion fails.